### ISSUE TITLE
[question] (Conan 2.0) on MSVC `conan profile detect` leads to runtime conflicts when using `-s build_type=Debug`

### ISSUE THREAD

**Issue by prince-chrismc** [conan-io/conan#13232]
### What is your question?

Using the default settings from `conan profile detect` for Windows.

Will give you the following on `conan install`

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release 👈 This here is populated for both profiles
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
```

However the default `runtime_type` profile plugin

```py
    settings = profile.settings
    if settings.get("compiler") == "msvc" and settings.get("compiler.runtime"):
        if settings.get("compiler.runtime_type") is None: # 👈  This only kicks in if the setting is missing -- which it never is
            runtime = "Debug" if settings.get("build_type") == "Debug" else "Release"
            try:
                settings["compiler.runtime_type"] = runtime
            except ConanException:
                pass
```

So the more simple use case `conan install -s build_type=Debug` leads to the profile helper not kicking in and mismatched sysmbols

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Debug 👈  This is the easy way
compiler=msvc
compiler.cppstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release 🚫 Does not match
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows
```

So it will result in 

```
Catch2d.lib(catch_session.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in Main.obj [D:\a\argon\argon\build\tests\test-argon.vcxproj]
Catch2d.lib(catch_session.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in Main.obj [D:\a\argon\argon\build\tests\test-argon.vcxproj]
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by ChrisThrasher** [comment]
Here is a CI pipeline run that illustrates this error.
https://github.com/ChrisThrasher/argon/actions/runs/4257358454

I was able to fix it by setting `-s compiler.runtime_type` to the correct config for the Windows CI jobs. Below is a link to a passing CI pipeline.
https://github.com/ChrisThrasher/argon/actions/runs/4266883423

The library in question has no dependencies. The reason I used Conan was to grab Catch2 which I only use in the test executable. The failing CI job is first installing the library then using `find_package` to consume that library as a way to test that the CMake install interface of the library works. The failure occurs when trying to build the project that consumes the library. Prior to using Conan this worked just fine but something about integrating Conan causes issues with MSVC iterator debug levels in the compiled library.
